subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdPlugin.java

```diff
@@ -180,14 +180,14 @@ private String calculateDefaultDependencyNotation(VersionNumber toolVersion) {
     }
 
     @Override
-    protected void configureForSourceSet(final SourceSet sourceSet, Pmd task) {
+    protected void configureForSourceSet(final SourceSet sourceSet, final Pmd task) {
         task.setDescription("Run PMD analysis for " + sourceSet.getName() + " classes");
         task.setSource(sourceSet.getAllJava());
         ConventionMapping taskMapping = task.getConventionMapping();
         taskMapping.map("classpath", new Callable<FileCollection>() {
             @Override
             public FileCollection call() throws Exception {
-                return sourceSet.getCompileClasspath();
+                return task.getProject().files(sourceSet.getOutput(), sourceSet.getCompileClasspath());
```

Comment:
This can be condensed a little. Both calls `sourceSet.getOutput()` and `sourceSet.getCompileClasspath()` return implementations of `FileCollection`. This interface as a convenient `plus(FileCollection fileCollection)` method that simply allows us to add them together:

    return sourceSet.getOutput().plus(sourceSet.getCompileClasspath());